clang/lib/CIR/CodeGen/CIRGenBuiltin.cpp

```diff
@@ -1462,8 +1462,19 @@ RValue CIRGenFunction::buildBuiltinExpr(const GlobalDecl GD, unsigned BuiltinID,
     llvm_unreachable("BI__builtin_wmemcmp NYI");
   case Builtin::BI__builtin_dwarf_cfa:
     llvm_unreachable("BI__builtin_dwarf_cfa NYI");
-  case Builtin::BI__builtin_return_address:
-    llvm_unreachable("BI__builtin_return_address NYI");
+  case Builtin::BI__builtin_return_address: {
+    mlir::Location loc = getLoc(E->getExprLoc());
```

Comment:
This one deserves its own CIR operation, it's not really a builtin that is just mapping LLVM behavior up, it's the opposite.

Reply:
I see, will do it. The reason I didn't introduce a CIR Op for this is that LLVM dialect doesn't have a ReturnAddress or FrameAddressOp, but it maybe still useful to expose the semantics to future CIR optimizer by having its own CIR Op.